src/translit.js

```diff
@@ -10,22 +10,41 @@ languages.shahmukhi = require('./translit_modules/shahmukhi');
  *
  * @since 1.0.0
  * @param {string} gurmukhi The string from to generate transliteration
+ * @param (optional) {string} language The name of language translit module to use
+ * @param (optional) {Map} map The name of language word map to use (for per-word translit)
  * @returns {string} Returns a string of text
  * @example
  *
  * translit('Awie imlu gurisK Awie imlu qU myry gurU ky ipAwry ]');
  * // => 'aai mil gursikh aai mil too mere guroo ke piaare ||'
  */
 
-function translit(gurmukhi, language = 'english') {
+function translit(gurmukhi, language = 'english', map = null) {
   if (language === 'all') {
     return Object.keys(languages).reduce((out, x) => {
       /* eslint no-param-reassign: 0 */
       out[x] = languages[x](gurmukhi);
       return out;
     }, {});
   }
+
+  if (map != null) {
+    return gurmukhi.split(' ').map((word) => {
+      // remove embedded vishraams before checking map
+      const vishraamRegex = /[.,;]$/;
+      const vishraam = word.match(vishraamRegex);
+      word = word.replace(vishraamRegex, '');
```

Comment:
this works but it **not** good. https://github.com/airbnb/javascript#functions--reassign-params. I will add a linter to enforce best practices.